crates/rpc/src/server/accept.rs

```diff
@@ -224,12 +224,16 @@ where
     }
 
     fn call(&mut self, request: http::Request<B>) -> Self::Future {
+        // Skip negotiation entirely for CORS preflight/non-gRPC requests.
+        if request.method() == http::Method::OPTIONS {
+            return self.inner.call(request).boxed();
+        }
```

Comment:
Another alternative would be to only have this layer active for `get/post` methods iiuc?

But I think this is fine.

Could you maybe expand on the comment with a _why_ this is necessary?

```suggestion
        // Skip negotiation entirely for CORS preflight/non-gRPC requests.
        //
        // Browsers often automatically perform an `OPTIONS` check _before_ the client
        // SDK can inject the appropriate `ACCEPT` header, causing a rejection.
        // Since an `OPTIONS` request does nothing its safe for us to simply allow them.
        if request.method() == http::Method::OPTIONS {
            return self.inner.call(request).boxed();
        }
```

Reply:
Addressed in [3bb0d00](https://github.com/0xMiden/miden-node/pull/1436/commits/3bb0d00b4efc4bb9ec22e476d56e862dc0df38cb)